static/main.js

```diff
@@ -35,7 +34,11 @@ $("#join-form").on("submit", ev => {
     headers: { "Content-Type": "application/json; charset=utf-8" },
     body: JSON.stringify(data)
   })
-    .then(_ => {
+    .then(response => {
+      if (response.status >= 400 && response.status < 600) {
```

Comment:
Huh, does the fetch API not reject on 4xx and 5xx?

Reply:
Yeah it doesn't actually error on error codes.